miro_preprocessor/image_sorter/src/sorter_logic.py

```diff
@@ -15,14 +77,55 @@ class Decision(enum.Enum):
     catalogue_api = 'catalogue_api'
 
 
-class Undecidable(Exception):
-    pass
-
-
 def sort_image(collection, image_data):
-    # For now reproduce the behaviour of the miro_adapter and send all images to the catalogue API.
-    return Decision.catalogue_api
+    # def _normalise_string(s):
+    #     if s is not None:
+    #         s = s.lower()
+    #     if s == "":
+    #         s = None
+    #     return s
+    #
+    # def _get(key):
+    #     return _normalise_string(image_data.get(key))
+    #
+    # def _compare(key, value):
+    #     return _get(key) == _normalise_string(value)
+    #
+    # def _search(regex, key):
+    #     return re.search(regex, _get(key))
+    #
+    # def _is_blank(key):
+    #     return _get(key) is None
+    #
+    # def _is_collection(collection_name):
+    #     return _normalise_string(collection) == f"source/images-{collection_name}"
+    #
+    # title_matches = [
+    #     _normalise_string("Awaiting description"),
+    #     _normalise_string("Awaiting removal from MIRO as a duplicate"),
+    #     _normalise_string("Awaiting captions and consent form"),
+    #     _normalise_string("Awaiting catalogue details"),
+    #     _normalise_string("Awaiting caption info"),
+    #     _normalise_string("No info available for this object"),
+    #     _normalise_string("No neg")
+    # ]
+    #
+    # c = {
+    #     "image_title is not one of title_matches": (
+    #         any(
+    #             [_compare("image_title", s) for s in title_matches]
+    #         ) or _search("-{1,2}", "image_title")
+    #     ),
+    # }
+
+    print(image_data)
+    r = Rules(collection, image_data)
+    print(r)
 
-    raise Undecidable(
-        f'No decision for collection={collection!r}, image_data={image_data!r}'
-    )
+    if r.is_f_collection or \
+            (r.is_l_or_m_or_v_collection and r.image_library_dept_is_Archives_and_Manuscripts) or \
+            (r.is_l_or_m_or_v_collection and r.image_tech_captured_mode_is_videodisc) or \
+            (r.is_l_or_m_or_v_collection and not r.is_innopac_id_8_digits and r.is_title_blank and r.is_image_pub_title_blank and r.is_image_pub_periodical_blank):
```

Comment:
Consider reversing this logic so that we don't need the not here so we can be clearer about what this is doing without worrying about operator precedence, i.e, make this `is_innopac_id__not_8_digits